<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>496</width>
    <height>704</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Transaction data generation Tool alpha 0.0.2</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="1" column="0">
    <widget class="QLabel" name="label_address">
     <property name="text">
      <string>Address of the receiver:</string>
     </property>
    </widget>
   </item>
   <item row="9" column="1">
    <widget class="QLineEdit" name="textbox_networkid">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>1</string>
     </property>
     <property name="readOnly">
      <bool>false</bool>
     </property>
     <property name="placeholderText">
      <string/>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="label_amount">
     <property name="toolTip">
      <string>select input method for network id</string>
     </property>
     <property name="text">
      <string>Amount:</string>
     </property>
    </widget>
   </item>
   <item row="7" column="1" rowspan="2">
    <widget class="QComboBox" name="combobox_networkid">
     <property name="toolTip">
      <string>https://ethereum.stackexchange.com/questions/17051/how-to-select-a-network-id-or-is-there-a-list-of-network-ids</string>
     </property>
     <property name="toolTipDuration">
      <number>20</number>
     </property>
     <item>
      <property name="text">
       <string extracomment="mainnet">1 - Mainnet</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>2 - Morden</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>3 - Ropsten</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>4 - Rinkeby</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>8 - Ubiq</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>42 - Kovan</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>77 - Sokol</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>99 - Core</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="label_gasprice">
     <property name="text">
      <string>Gas price per gas unit:</string>
     </property>
    </widget>
   </item>
   <item row="6" column="0">
    <widget class="QLabel" name="label_data">
     <property name="text">
      <string>Additional data:</string>
     </property>
    </widget>
   </item>
   <item row="5" column="1" colspan="4">
    <widget class="QLineEdit" name="textbox_gas">
     <property name="inputMethodHints">
      <set>Qt::ImhDigitsOnly</set>
     </property>
     <property name="text">
      <string>21000</string>
     </property>
     <property name="placeholderText">
      <string>Default: 21000</string>
     </property>
    </widget>
   </item>
   <item row="6" column="1" colspan="4">
    <widget class="QLineEdit" name="textbox_data">
     <property name="text">
      <string/>
     </property>
     <property name="placeholderText">
      <string>If used, don't forget to adapt gas amount</string>
     </property>
    </widget>
   </item>
   <item row="15" column="1">
    <widget class="QLineEdit" name="textbox_privatekey">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
     <property name="placeholderText">
      <string>Enter private key (hex)</string>
     </property>
    </widget>
   </item>
   <item row="11" column="1">
    <widget class="QLineEdit" name="textbox_keystore">
     <property name="text">
      <string/>
     </property>
     <property name="placeholderText">
      <string>Enter filepath to keystore</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="label_nonce">
     <property name="text">
      <string>Current nonce value:</string>
     </property>
    </widget>
   </item>
   <item row="5" column="0">
    <widget class="QLabel" name="label_gas">
     <property name="text">
      <string>Gas amount to include:</string>
     </property>
    </widget>
   </item>
   <item row="18" column="0" colspan="5">
    <widget class="QTextBrowser" name="textbrowser_result">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="html">
      <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Noto Sans'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="placeholderText">
      <string>This box will contain your transaction data :)</string>
     </property>
    </widget>
   </item>
   <item row="2" column="1" colspan="3">
    <widget class="QLineEdit" name="textbox_amount">
     <property name="whatsThis">
      <string/>
     </property>
     <property name="accessibleName">
      <string/>
     </property>
     <property name="accessibleDescription">
      <string/>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhLatinOnly</set>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="maxLength">
      <number>42</number>
     </property>
     <property name="placeholderText">
      <string>Amount to send. Check the unit</string>
     </property>
    </widget>
   </item>
   <item row="11" column="0">
    <widget class="QLabel" name="label_keystore">
     <property name="text">
      <string>Keystore JSON file:</string>
     </property>
    </widget>
   </item>
   <item row="2" column="4">
    <widget class="QComboBox" name="combobox_unit">
     <item>
      <property name="text">
       <string>ether</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>finney (milli)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>szabo (micro)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>shannon (nano)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>lovelace (pico)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>babbage (femto)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>wei (atto)</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="7" column="0" rowspan="2">
    <widget class="QLabel" name="label_networkid">
     <property name="text">
      <string>Network ID:</string>
     </property>
    </widget>
   </item>
   <item row="4" column="1" colspan="4">
    <widget class="QLineEdit" name="textbox_gasprice">
     <property name="inputMethodHints">
      <set>Qt::ImhDigitsOnly</set>
     </property>
     <property name="text">
      <string>4000000000</string>
     </property>
     <property name="placeholderText">
      <string>Input in WEI (default: 4 GWei)</string>
     </property>
    </widget>
   </item>
   <item row="1" column="1" colspan="4">
    <widget class="QLineEdit" name="textbox_receiver">
     <property name="whatsThis">
      <string/>
     </property>
     <property name="accessibleName">
      <string/>
     </property>
     <property name="accessibleDescription">
      <string/>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhLatinOnly</set>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="maxLength">
      <number>42</number>
     </property>
     <property name="placeholderText">
      <string>Ethereum address of the receiver (0x...)</string>
     </property>
    </widget>
   </item>
   <item row="15" column="0">
    <widget class="QLabel" name="label_privatekey">
     <property name="text">
      <string>Private key:</string>
     </property>
    </widget>
   </item>
   <item row="3" column="1" colspan="4">
    <widget class="QLineEdit" name="textbox_nonce">
     <property name="text">
      <string/>
     </property>
     <property name="placeholderText">
      <string>Check your last outgoing transaction</string>
     </property>
    </widget>
   </item>
   <item row="16" column="0" colspan="5">
    <widget class="QPushButton" name="pushbutton_generate">
     <property name="font">
      <font>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Generate transacton data</string>
     </property>
    </widget>
   </item>
   <item row="14" column="1">
    <widget class="QLineEdit" name="textbox_keystore_pw">
     <property name="inputMethodHints">
      <set>Qt::ImhHiddenText|Qt::ImhNoAutoUppercase|Qt::ImhNoPredictiveText|Qt::ImhSensitiveData</set>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="echoMode">
      <enum>QLineEdit::Password</enum>
     </property>
     <property name="readOnly">
      <bool>false</bool>
     </property>
     <property name="placeholderText">
      <string>Keystore password</string>
     </property>
    </widget>
   </item>
   <item row="7" column="4">
    <widget class="QRadioButton" name="radiobutton_networkid_dropdown">
     <property name="acceptDrops">
      <bool>false</bool>
     </property>
     <property name="toolTip">
      <string>select input method for network id</string>
     </property>
     <property name="text">
      <string>Dropdown menu</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
     <property name="autoRepeat">
      <bool>false</bool>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">group_radio_networkid</string>
     </attribute>
    </widget>
   </item>
   <item row="9" column="4">
    <widget class="QRadioButton" name="radiobutton_networkid_textbox">
     <property name="acceptDrops">
      <bool>false</bool>
     </property>
     <property name="toolTip">
      <string>select input method for network id</string>
     </property>
     <property name="text">
      <string>Textbox</string>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
     <property name="autoRepeat">
      <bool>false</bool>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">group_radio_networkid</string>
     </attribute>
    </widget>
   </item>
   <item row="15" column="4">
    <widget class="QRadioButton" name="radiobutton_key">
     <property name="text">
      <string>Private key</string>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">group_radio_key</string>
     </attribute>
    </widget>
   </item>
   <item row="14" column="0">
    <widget class="QLabel" name="label_keystore_pw">
     <property name="text">
      <string>Keystore password:</string>
     </property>
    </widget>
   </item>
   <item row="14" column="4">
    <widget class="QRadioButton" name="radiobutton_keystore">
     <property name="text">
      <string>Keystore</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">group_radio_key</string>
     </attribute>
    </widget>
   </item>
   <item row="11" column="4">
    <widget class="QPushButton" name="pushbutton_browse">
     <property name="text">
      <string>Browse</string>
     </property>
    </widget>
   </item>
   <item row="19" column="0">
    <widget class="QLabel" name="label_address_text">
     <property name="text">
      <string>transaction data valid for:</string>
     </property>
    </widget>
   </item>
   <item row="19" column="1" colspan="2">
    <widget class="QLabel" name="label_address_sender">
     <property name="text">
      <string/>
     </property>
     <property name="textInteractionFlags">
      <set>Qt::LinksAccessibleByMouse|Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="group_radio_networkid"/>
  <buttongroup name="group_radio_key"/>
 </buttongroups>
</ui>
